src/emailmessagelistmodel.cpp

```diff
@@ -337,9 +337,9 @@ void EmailMessageListModel::setFolderKey(int id, QMailMessageKey messageKey)
     if (!m_currentFolderId.isValid())
         return;
     // Local folders (e.g outbox) can have messages from several accounts.
-    QMailMessageKey accountKey = QMailMessageKey::parentAccountId(m_mailAccountIds) & messageKey;
+    QMailMessageKey accountKey = QMailMessageKey::parentAccountId(m_mailAccountIds);
     QMailMessageKey folderKey = accountKey & QMailMessageKey::parentFolderId(m_currentFolderId);
-    QMailMessageListModel::setKey(folderKey);
+    QMailMessageListModel::setKey(folderKey & messageKey);
```

Comment:
No, I got also confused, so decided to change it since was mixed above with the account key.